tests/sls/homu/valid_repos.py

```diff
@@ -0,0 +1,59 @@
+import toml
+import requests
+from tests.util import Failure, Success
+
+
+def repo_pager(endpoint, repoblock, auth):
+    """
+    recursively consumes an endpoint and an empty (self-populating) list
+    it will exit when it fails to find the 'next' key in the links dict
+    """
+    req = requests.get(endpoint, headers=auth)
```

Comment:
If the endpoint returns anything about failed auth, please bail out right here with a descriptive error message.